src/NServiceBus.Core/Transports/Msmq/QueuePermissions.cs

```diff
@@ -19,14 +19,18 @@ public void CheckQueuePermissions(IReadOnlyCollection<string> queues)
         static void CheckQueue(string address)
         {
             var msmqAddress = MsmqAddress.Parse(address);
-            var queuePath = msmqAddress.PathWithoutPrefix;
 
-            if (MessageQueue.Exists(queuePath))
+            MessageQueue messageQueue;
+            if (!MsmqUtilities.TryOpenQueue(msmqAddress, out messageQueue))
             {
-                using (var messageQueue = new MessageQueue(queuePath))
-                {
-                    WarnIfPublicAccess(messageQueue);
-                }
+                Logger.Warn($"Unable to open the queue at address '{address}'. Make sure the queue exists, and the address is correct.");
+                Logger.Debug("Processing will still continue.");
```

Comment:
@bording yeah, those are copied verbatim from QueueCreator, I just moved this method here.